### PR TITLE
Cloudshell account settings ter (#1)

### PR DESCRIPTION
**Pull request by hartyt**
Cloud Shell account settings resource and datasource. 
**Note**: I used openapi-sdkgen to generate the resource & datasource, I followed the instructions, manually modified the tests and documentation. Cloud Shell public API & SDKs can be found here https://cloud.ibm.com/apidocs/cloudshell

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMCloudShellAccountSettings'
=== RUN   TestAccIBMCloudShellAccountSettingsDataSourceBasic
--- PASS: TestAccIBMCloudShellAccountSettingsDataSourceBasic (22.77s)
=== RUN   TestAccIBMCloudShellAccountSettingsDataSourceAllArgs
--- PASS: TestAccIBMCloudShellAccountSettingsDataSourceAllArgs (29.57s)
=== RUN   TestAccIBMCloudShellAccountSettingsBasic
--- PASS: TestAccIBMCloudShellAccountSettingsBasic (27.64s)
=== RUN   TestAccIBMCloudShellAccountSettingsAllArgs
--- PASS: TestAccIBMCloudShellAccountSettingsAllArgs (41.01s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm	124.130s
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/internal/hashcode	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/internal/mutexkv	(cached) [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/version	[no test files]
...
```

### CODE REVIEW

File: cloud-docs/ibm-cloud-shell-data-sources.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
```
[the remark this answers is not in the document]
Comment (hartyt) [reply cloud-docs/ibm-cloud-shell-data-sources.md:1]: this was generated by the Terraform-Provider-Generation tool, should i still remove it?

File: examples/ibm-cloud-shell/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Example for IBMCloudShellV1
```
[the remark this answers is not in the document]
Comment (hartyt) [reply examples/ibm-cloud-shell/README.md:1]: can you clarify this one a bit more? Do you want me to remove V1 from the name altogether? Looking at some of the other examples they also have V1 in the name. Again the Terraform-Provider-Generation tool used this naming convention

File: ibm/data_source_ibm_cloud_shell_account_settings.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	return &schema.Resource{
+		ReadContext: dataSourceIBMCloudShellAccountSettingsRead,
+
+		Schema: map[string]*schema.Schema{
+			"account_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The account ID in which the account settings belong to.",
+			},
+			"id": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (hartyt) [reply ibm/data_source_ibm_cloud_shell_account_settings.go:27]: the `dataSourceIBMCloudShellAccountSettingsRead` explicitly set the id using `d.SetId(fmt.Sprintf("%s", *getAccountSettingsOptions.AccountID))`
if i remove the id from the schema then `d.SetId(fmt.Sprintf("%s", *getAccountSettingsOptions.AccountID))` will fail

File: cloud-docs/ibm-cloud-shell-data-sources.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkantare) [reply cloud-docs/ibm-cloud-shell-data-sources.md:1]: Yes those docs should be removed..We already have docs in website floder

File: ibm/data_source_ibm_cloud_shell_account_settings.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (hkantare) [reply ibm/data_source_ibm_cloud_shell_account_settings.go:27]: Yes all the resources and datasource will set the ID..We don't need to explictly defined in the scheam. Just remove the definitation of id from schema

File: examples/ibm-cloud-shell/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hkantare) [reply examples/ibm-cloud-shell/README.md:1]: Ok I just looked other examples this should be fine

File: ibm/data_source_ibm_cloud_shell_account_settings.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (hartyt) [reply ibm/data_source_ibm_cloud_shell_account_settings.go]: done

File: ibm/resource_ibm_cloud_shell_account_settings.go [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+						},
+						"key": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Name of the region.",
+						},
+					},
+				},
+			},
+			"id": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (hartyt) [reply ibm/resource_ibm_cloud_shell_account_settings.go]: done

File: ibm/resource_ibm_cloud_shell_account_settings.go [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+		updateAccountSettingsOptions.SetRegions(regions)
+	}
+
+	accountSettings, response, err := ibmCloudShellClient.UpdateAccountSettingsWithContext(context, updateAccountSettingsOptions)
+	if err != nil {
+		log.Printf("[DEBUG] UpdateAccountSettingsWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("UpdateAccountSettingsWithContext failed %s\n%s", err, response))
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", *updateAccountSettingsOptions.AccountID, *accountSettings.AccountID))
```
[the remark this answers is not in the document]
Comment (hartyt) [reply ibm/resource_ibm_cloud_shell_account_settings.go]: done, I changed this to `d.SetId(*accountSettings.ID)`

File: website/docs/r/cloud_shell_account_settings.html.markdown [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `account_id` - (Required, Forces new resource, string) The account ID in which the account settings belong to.
+* `rev` - (Optional, string) Unique revision number for the settings object.
```
[the remark this answers is not in the document]
Comment (hartyt) [reply website/docs/r/cloud_shell_account_settings.html.markdown]: done

File: ibm/config.go [hunk 7]
```diff
@@ -2066,6 +2076,19 @@ func (c *Config) ClientSession() (interface{}, error) {
 	}
 	session.resourceManagerAPI = resourceManagerClient
 
+	ibmCloudShellClientOptions := &ibmcloudshellv1.IBMCloudShellV1Options{
```
Comment (hkantare) [inline ibm/config.go:2079]: Can you please provide a environmental variable to set to stage environmental and support for private(if supported)...please see other client intialization
```
apicurl := contructEndpoint(fmt.Sprintf("api.%s.apigw", c.Region), fmt.Sprintf("%s/controller", cloudEndpoint))
	if c.Visibility == "private" || c.Visibility == "public-and-private" {
		apicurl = contructEndpoint(fmt.Sprintf("api.private.%s.apigw", c.Region), fmt.Sprintf("%s/controller", cloudEndpoint))
	}
	APIGatewayControllerAPIV1Options := &apigateway.ApiGatewayControllerApiV1Options{
		URL:           envFallBack([]string{"IBMCLOUD_API_GATEWAY_ENDPOINT"}, apicurl),
		Authenticator: &core.NoAuthAuthenticator{},
	}
	apigatewayAPI, err := apigateway.NewApiGatewayControllerApiV1(APIGatewayControllerAPIV1Options)
	if err != nil {
		session.apigatewayErr = fmt.Errorf("Error occured while configuring  APIGateway service: %q", err)
	}
	session.apigatewayAPI = apigatewayAPI
	```

File: website/docs/r/cloud_shell_account_settings.html.markdown [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+* `updated_at` - Timestamp of last update in Unix epoch time.
+* `updated_by` - IAM ID of last updater.
+
+## Import
+
+You can import the `cloud_shell_account_settings` resource by using `account_id`.
+The `account_id` property can be formed from `account_id`, and `account_id` in the following format:
+
+```
+<account_id>/<account_id>
```
Comment (hkantare) [inline website/docs/r/cloud_shell_account_settings.html.markdown:88]: i think here we need only one account_id after u changed in resource

File: ibm/config.go (hunk 7 above)
Comment (hartyt) [reply ibm/config.go:2079]: we do not currently support private endpoints

File: website/docs/r/cloud_shell_account_settings.html.markdown (hunk 8 above)
Comment (hartyt) [reply website/docs/r/cloud_shell_account_settings.html.markdown]: done

File: ibm/config.go (hunk 7 above)
Comment (hartyt) [reply ibm/config.go:2079]: done

**Pull request merged**